vnext/ReactUWP/Utils/BaseScriptStoreImpl.cpp

```diff
@@ -0,0 +1,341 @@
+#include "pch.h"
+
+#include "BaseScriptStoreImpl.h"
+
+#include <fstream>
+
+#include <winrt/Windows.Storage.FileProperties.h>
+#include <winrt/Windows.Storage.Streams.h>
+
+#include <codecvt>
+#include <locale>
+
+using namespace facebook;
+
+namespace winrt {
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Storage;
+}; // namespace winrt
+
+namespace react {
+namespace uwp {
+
+namespace {
+
+class ByteArrayBuffer : public facebook::jsi::Buffer
+{
+public:
+  size_t size() const override
+  {
+    return size_;
+  }
+
+  const uint8_t* data() const
+  {
+    return byteArray_.get();
+  }
+
+  uint8_t* data()
+  {
+    return byteArray_.get();
+  }
+
+  ByteArrayBuffer(size_t size)
+    : size_(size), byteArray_(std::make_unique<uint8_t[]>(size)) {}
+
+  ByteArrayBuffer(ByteArrayBuffer&&) = default;
+  ByteArrayBuffer& operator=(ByteArrayBuffer&&) = default;
+
+private:
+  ByteArrayBuffer(const ByteArrayBuffer&) = delete;
+  ByteArrayBuffer& operator=(const ByteArrayBuffer&) = delete;
+
+  size_t size_;
+  std::unique_ptr<uint8_t[]> byteArray_;
+};
+
+class BufferViewBuffer : public facebook::jsi::Buffer
+{
+public:
+  size_t size() const override
+  {
+    return size_;
+  }
+
+  const uint8_t* data() const
+  {
+    return buffer_->data() + offset_;
+  }
+
+  uint8_t* data()
+  {
+    return const_cast<uint8_t*>(buffer_->data()) + offset_;
+  }
+
+  BufferViewBuffer(std::unique_ptr<const facebook::jsi::Buffer> buffer, size_t offset, size_t size)
+    : buffer_(std::move(buffer)), offset_(offset), size_(size)
+  {
+    if (size_ > buffer_->size() - offset)
+      std::terminate();
+  }
+
+  BufferViewBuffer(BufferViewBuffer&&) = default;
+  BufferViewBuffer& operator=(BufferViewBuffer&&) = default;
+
+private:
+  BufferViewBuffer(const BufferViewBuffer&) = delete;
+  BufferViewBuffer& operator=(const BufferViewBuffer&) = delete;
+
+  std::unique_ptr<const facebook::jsi::Buffer> buffer_;
+  size_t offset_;
+  size_t size_;
+};
+
+constexpr const char* PERSIST_MAGIC = "CHAKRAPREP";
```

Comment:
NIT: this isnt chakra specific. maybe a different constant for PERSIST_MAGIC?

Reply:
I've just renamed it to RNWPREP to be more generic.. :)

Reply:
thanks!